copy_test.go

```diff
@@ -136,9 +136,11 @@ func TestWriteError(t *testing.T) {
 	errCh := make(chan error, 1)
 	stop := uint32(0)
 	buf := make([]byte, 1000)
-	doCopy(dst, src, buf, errCh, &stop)
+	nw := 0
+	doCopy(dst, src, buf, errCh, &stop, func(n int) { nw += n })
 	reportedErr := <-errCh
 	assert.Contains(t, reportedErr.Error(), "use of closed network connection")
+	assert.Zero(t, nw, "Shouldn't have written any bytes")
```

Comment:
wouldn't hurt to add a test that is does count the correct number of bytes when copy works?